From aa9f721681601afc06dc9f28db2115971f3f38f5 Mon Sep 17 00:00:00 2001
From: Alex Van Damme <atv@google.com>
Date: Thu, 15 Aug 2024 16:46:40 -0700
Subject: [PATCH 2/5] SystemC support for verilator

---
 dependency_support/verilator/verilator.BUILD | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/dependency_support/verilator/verilator.BUILD b/dependency_support/verilator/verilator.BUILD
index 371a1dd..cd0c3fa 100644
--- a/dependency_support/verilator/verilator.BUILD
+++ b/dependency_support/verilator/verilator.BUILD
@@ -196,10 +196,12 @@ cc_library(
         "include/gtkwave/lz4.h",
         "include/gtkwave/wavealloca.h",
         "include/verilated.cpp",
         "include/verilated_fst_c.cpp",
+        "include/verilated_fst_sc.cpp",
         "include/verilated_imp.h",
         "include/verilated_syms.h",
         "include/verilated_vcd_c.cpp",
+        "include/verilated_vcd_sc.cpp",
     ],
     hdrs = [
         "include/verilated.h",
@@ -215,13 +217,20 @@ cc_library(
         # Needed for verilated_vcd_c.cpp and verilated_fst_c.cpp
         "include/verilated_trace_imp.h",
+        "include/verilated_fst_sc.h",
         "include/verilated_vcd_c.h",
+        "include/verilated_vcd_sc.h",
         "include/verilatedos.h",
         "include/verilated_types.h",
         "include/verilated_funcs.h",
     ],
     # TODO: Remove these once upstream fixes these warnings
     copts = ["-Wno-unused-const-variable"],
-    includes = ["include"],
+    includes = [
+        "include",
+        # include systemc header
+        "systemc/include",
+    ],
+    defines = ["VM_SC"],
     strip_include_prefix = "include/",
     textual_hdrs = [
         "include/gtkwave/fastlz.c",
-- 
2.46.0.184.g6999bdac58-goog

