From b47920432c8d4111224903d9bc5e7bb3a97619ce Mon Sep 17 00:00:00 2001
From: Alex Van Damme <atv@google.com>
Date: Thu, 15 Aug 2024 16:47:07 -0700
Subject: [PATCH 3/5] Add systemc lib support

---
 dependency_support/verilator/verilator.BUILD | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dependency_support/verilator/verilator.BUILD b/dependency_support/verilator/verilator.BUILD
index cd0c3fa..ea5cc41 100644
--- a/dependency_support/verilator/verilator.BUILD
+++ b/dependency_support/verilator/verilator.BUILD
@@ -239,6 +239,7 @@ cc_library(
     ],
     visibility = ["//visibility:public"],
     deps = [
+        "@accellera_systemc//:systemc",
         "@net_zlib//:zlib",
     ],
 )
-- 
2.46.0.184.g6999bdac58-goog

