From a9a8f13130ae94b50b0a17c7324ef2bba80411ab Mon Sep 17 00:00:00 2001
From: Alex Van Damme <atv@google.com>
Date: Thu, 20 Mar 2025 11:29:56 -0700
Subject: [PATCH 4/4] [Verilator] Enable public-flat-rw

---
 verilator/defs.bzl | 1 +
 1 file changed, 1 insertion(+)

diff --git a/verilator/defs.bzl b/verilator/defs.bzl
index fc8af53..d81bfc4 100644
--- a/verilator/defs.bzl
+++ b/verilator/defs.bzl
@@ -130,6 +130,7 @@ def _verilator_cc_library(ctx):
     args.add("--Mdir", verilator_output.path)
     args.add("--top-module", ctx.attr.module_top)
     args.add("--prefix", prefix)
+    args.add("--public-flat-rw")
 
     if ctx.attr.trace:
         args.add("--trace-fst")
-- 
2.49.0.395.g12beb8f557-goog

