From f1711f492eab6978b659e83e336eea421334d96e Mon Sep 17 00:00:00 2001
From: Alex Van Damme <atv@google.com>
Date: Thu, 1 May 2025 16:46:01 -0700
Subject: [PATCH 4/4] More jobs for cocotb

---
 cocotb/cocotb_wrapper.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cocotb/cocotb_wrapper.py b/cocotb/cocotb_wrapper.py
index d6829c6..3929519 100644
--- a/cocotb/cocotb_wrapper.py
+++ b/cocotb/cocotb_wrapper.py
@@ -13,6 +13,7 @@
 # limitations under the License.
 
 import argparse
+import cocotb_tools
 from cocotb_tools.runner import get_runner
 
 
@@ -184,6 +185,7 @@ if __name__ == "__main__":
         test_args += test_arg.split(" ")
     test_flags['test_args'] = test_args
 
+    cocotb_tools.runner.MAX_PARALLEL_BUILD_JOBS = 32
     runner = get_runner(args.sim)
     runner.build(**build_flags)
     results_xml = runner.test(**test_flags)
-- 
2.49.0.906.g1f30a19c02-goog

