From f7be24e2ceb3d0ab06f5339689d8a9b70e1d2607 Mon Sep 17 00:00:00 2001
From: Matthew Wilson <mwilson@google.com>
Date: Wed, 12 Nov 2025 15:06:02 -0800
Subject: [PATCH 7/7] Suppress skywater pdk loading

---
 dependency_support/dependency_support.bzl | 2 --
 1 file changed, 2 deletions(-)

diff --git a/dependency_support/dependency_support.bzl b/dependency_support/dependency_support.bzl
index a82d199..402b670 100644
--- a/dependency_support/dependency_support.bzl
+++ b/dependency_support/dependency_support.bzl
@@ -35,7 +35,6 @@ load("@rules_hdl//dependency_support/com_google_absl:com_google_absl.bzl", "com_
 load("@rules_hdl//dependency_support/com_google_googletest:com_google_googletest.bzl", "com_google_googletest")
 load("@rules_hdl//dependency_support/com_google_ortools:com_google_ortools.bzl", "com_google_ortools")
 load("@rules_hdl//dependency_support/com_google_protobuf:com_google_protobuf.bzl", "com_google_protobuf")
-load("@rules_hdl//dependency_support/com_google_skywater_pdk:com_google_skywater_pdk.bzl", "com_google_skywater_pdk")
 load("@rules_hdl//dependency_support/com_icarus_iverilog:com_icarus_iverilog.bzl", "com_icarus_iverilog")
 load("@rules_hdl//dependency_support/com_opencircuitdesign_magic:com_opencircuitdesign_magic.bzl", "com_opencircuitdesign_magic")
 load("@rules_hdl//dependency_support/com_opencircuitdesign_netgen:com_opencircuitdesign_netgen.bzl", "com_opencircuitdesign_netgen")
@@ -90,7 +89,6 @@ def dependency_support(register_toolchains = True):
     com_google_googletest()
     com_google_protobuf()
     com_google_ortools()
-    com_google_skywater_pdk()
     com_icarus_iverilog()
     com_opencircuitdesign_magic()
     com_opencircuitdesign_netgen()
-- 
2.52.0.rc1.455.g30608eb744-goog

